### ISSUE TITLE
Error when evaluating a sum of operations with arrays

### ISSUE THREAD

**Issue by oriolcg** [gridap/Gridap.jl#512]
I'm getting the following error when evaluating a sum of cell integrals involving operation with arrays:
```
ERROR: MethodError: no method matching zero(::Type{Array{Float64,1}})
Closest candidates are:
  zero(::Type{Missing}) at missing.jl:103
  zero(::Type{Dates.Time}) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.5\Dates\src\types.jl:406
  zero(::Type{Dates.DateTime}) at D:\buildbot\worker\package_win64\build\usr\share\julia\stdlib\v1.5\Dates\src\types.jl:404
  ...
```

The MWE is the following:
```julia
using Gridap

model = CartesianDiscreteModel((0,1,0,1),(2,2))
Ω = Triangulation(model)
dΩ = Measure(Ω,1)
reffeᵤ = ReferenceFE(lagrangian,Float64,1)
Vₕ = TestFESpace(model,reffeᵤ,conformity=:H1)
vₕ = get_cell_shapefuns(Vₕ)

α=[1.0 2.0 3.0 4.0]
α_Ω = CellField(α,Ω,ReferenceDomain())
vα_Ω = ∑(∫(α_Ω*vₕ)dΩ)
```

And the staktrace:
```
Stacktrace:
 [1] _sum_lazy_array(::Tuple{Tuple{Nothing,Gridap.Arrays.CachedArray{Float64,1,Array{Float64,1}},Tuple{Tuple{Tuple{Nothing,Gridap.Arrays.CachedArray{Float64,2,Array{Float64,2}},Tuple{Tuple{Tuple{Tuple{Tuple{Nothing,Nothing,Tuple{Nothing}},Gridap.Arrays.IndexItemPair{Int64,Gridap.Fields.ConstantField{Float64}}},Gridap.Arrays.CachedArray{Float64,1,Array{Float64,1}},Tuple{Nothing}},Gridap.Arrays.IndexItemPair{Int64,Array{Float64,1}}},Nothing}},Gridap.Arrays.IndexItemPair{Int64,Array{Float64,2}}},Nothing,Nothing}},Gridap.Arrays.IndexItemPair{Int64,Array{Float64,1}}}, ::Gridap.Arrays.LazyArray{FillArrays.Fill{Gridap.Fields.IntegrationMap,1,Tuple{Base.OneTo{Int64}}},Array{Float64,1},1,Tuple{Gridap.Arrays.LazyArray{FillArrays.Fill{Gridap.Fields.BroadcastingFieldOpMap{typeof(*)},1,Tuple{Base.OneTo{Int64}}},Array{Float64,2},1,Tuple{Gridap.Arrays.LazyArray{Gridap.Arrays.LazyArray{FillArrays.Fill{UnionAll,2,Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}},Gridap.Fields.ConstantField{Float64},2,Tuple{Array{Float64,2}}},Array{Float64,1},1,Tuple{FillArrays.Fill{Array{VectorValue{2,Float64},1},1,Tuple{Base.OneTo{Int64}}}}},FillArrays.Fill{Array{Float64,2},1,Tuple{Base.OneTo{Int64}}}}},FillArrays.Fill{Array{Float64,1},1,Tuple{Base.OneTo{Int64}}},FillArrays.Fill{Array{TensorValue{2,2,Float64,4},1},1,Tuple{Base.OneTo{Int64}}}}}) at C:\Users\ocolomesgene\.julia\packages\Gridap\g8tlR\src\Arrays\LazyArrays.jl:208
 [2] sum(::Gridap.Arrays.LazyArray{FillArrays.Fill{Gridap.Fields.IntegrationMap,1,Tuple{Base.OneTo{Int64}}},Array{Float64,1},1,Tuple{Gridap.Arrays.LazyArray{FillArrays.Fill{Gridap.Fields.BroadcastingFieldOpMap{typeof(*)},1,Tuple{Base.OneTo{Int64}}},Array{Float64,2},1,Tuple{Gridap.Arrays.LazyArray{Gridap.Arrays.LazyArray{FillArrays.Fill{UnionAll,2,Tuple{Base.OneTo{Int64},Base.OneTo{Int64}}},Gridap.Fields.ConstantField{Float64},2,Tuple{Array{Float64,2}}},Array{Float64,1},1,Tuple{FillArrays.Fill{Array{VectorValue{2,Float64},1},1,Tuple{Base.OneTo{Int64}}}}},FillArrays.Fill{Array{Float64,2},1,Tuple{Base.OneTo{Int64}}}}},FillArrays.Fill{Array{Float64,1},1,Tuple{Base.OneTo{Int64}}},FillArrays.Fill{Array{TensorValue{2,2,Float64,4},1},1,Tuple{Base.OneTo{Int64}}}}}) at C:\Users\ocolomesgene\.julia\packages\Gridap\g8tlR\src\Arrays\LazyArrays.jl:204
 [3] iterate at .\generator.jl:47 [inlined]
 [4] collect(::Base.Generator{Base.ValueIterator{IdDict{Triangulation,AbstractArray}},typeof(sum)}) at .\array.jl:686
 [5] map at .\abstractarray.jl:2188 [inlined]
 [6] sum(::DomainContribution) at C:\Users\ocolomesgene\.julia\packages\Gridap\g8tlR\src\CellData\DomainContributions.jl:76
...
```

Any clue on what could I be doing wrong?

**Comment by fverdugo** [comment]
Hi @oriolcg!

I can reproduce the error.  It should be fixed by replacing this line
https://github.com/gridap/Gridap.jl/blob/0aa825475b3f1c116249fcf7354adae3d9d2b92d/src/Arrays/LazyArrays.jl#L208
with
```julia
  r = zero(testitem(a))
```

Can you file a PR with this?

As a side note,  I am wandering if it makes sense to compute this quantity `∑(∫(α_Ω*vₕ)dΩ)`.  E.g., summing 
 these elemental vectors is only possible when all cells have the same number of DOFs.  This `∑(∫(vₕ)dΩ)` works since `∫(vₕ)dΩ`  is a `Fill` array in this particular case, but it would also fail in other cases .

**Comment by oriolcg** [comment]
Thanks @fverdugo, I'll submit a PR with the fix.

> As a side note, I am wandering if it makes sense to compute this quantity `∑(∫(α_Ω*vₕ)dΩ)`

This was just an expression for the MWE. It all appeared when working with a bilinear form with piece-wise constant coefficients that are discontinuous across elements given by an array instead of a function.

**Comment by fverdugo** [comment]
Closed via PR #513
